packages/app/src/app/pages/Sandbox/Editor/Navigation/index.tsx

```diff
@@ -44,26 +44,23 @@ type IconProps = {
   item: INavigationItem;
   isDisabled?: boolean;
   singleton: TippyProps['singleton'];
+  selected: boolean;
+  hasChanges: boolean;
+  hasConflicts: boolean;
+  isFetching: boolean;
 };
 const IconComponent: FunctionComponent<IconProps> = ({
   item: { id, name },
   isDisabled,
   singleton,
+  selected,
+  hasChanges,
+  hasConflicts,
+  isFetching,
 }) => {
   const { setWorkspaceHidden, setWorkspaceItem } = useActions().workspace;
-  const {
-    workspace: { openedWorkspaceItem, workspaceHidden },
-    git: { gitChanges, isFetching, conflicts },
-  } = useAppState();
 
   const Icon = IDS_TO_ICONS[id];
-  const selected = !workspaceHidden && id === openedWorkspaceItem;
-  const hasChanges =
```

Comment:
Is this file change also intended to be here? Just to make sure something else didn't leak into this PR 😄 .

Reply:
Yes, this is. This related to an improvement to avoid rerender in a component, so what I have done here was move the dependencies a level up, so then React is able to leverage the props comparations to avoid unnecessaries renders.

Reply:
Great!!